rasa/nlu/data_augmentation/augmenter.py

```diff
@@ -0,0 +1,699 @@
+        )
+        recall_change = intent_results["recall"] - intent_results_original["recall"]
+        f1_change = intent_results["f1-score"] - intent_results_original["f1-score"]
+
+        intent_results["precision_change"] = intent_summary[intent][
+            "precision_change"
+        ] = precision_change
+        intent_results["recall_change"] = intent_summary[intent][
+            "recall_change"
+        ] = recall_change
+        intent_results["f1-score_change"] = intent_summary[intent][
+            "f1-score_change"
+        ] = f1_change
+        intent_report[intent] = intent_results
+
+    return (intent_summary, intent_report)
+
+
+def _create_summary_report(
+    intent_report: Dict[Text, float],
+    classification_report: Dict[Text, Dict[Text, Any]],
+    training_intents: List[Text],
+    pooled_intents: Set[Text],
+    output_directory: Text,
+) -> Dict[Text, Dict[Text, float]]:
+    """Creates a summary of the effect of data augmentation.
+
+    Args:
+        intent_report: Report of the model run *with* data augmentation.
+        classification_report: Classification report of the model run *without* data augmentation.
+        training_intents: All intents in the training data (non-augmented).
+        pooled_intents: The intents that have been selected for data augmentation.
+        output_directory: Directory to store the output reports in.
+
+    Returns:
+        Tuple representing the data augmentation summary as well as the set of changed intents.
+    """
+
+    # Retrieve intents for which performance has changed
+    changed_intents = (
+        _get_intents_with_performance_changes(
+            classification_report, intent_report, training_intents,
+        )
+        - pooled_intents
+    )
+
+    # Create and update result reports
+    report_tuple = _create_augmentation_summary(
+        pooled_intents, changed_intents, classification_report, intent_report,
+    )
+
+    intent_summary = report_tuple[0]
+    intent_report.update(report_tuple[1])
+
+    # Store report to file
+    rasa.shared.utils.io.dump_obj_as_json_to_file(
+        os.path.join(output_directory, "intent_report.json"), intent_report,
+    )
+
+    return intent_summary
+
+
+def _plot_summary_report(
+    intent_summary: Dict[Text, Dict[Text, float]], output_directory: Text,
+) -> None:
+    """Plots the data augmentation summary.
+
+    Args:
+        intent_summary: Summary report of the effect of data augmentation.
+        output_directory: Directory to store the summary plot in.
+    """
+    for metric in ["precision", "recall", "f1-score"]:
+        output_file = os.path.join(output_directory, f"{metric}_changes.png")
+        rasa.utils.plotting.plot_intent_augmentation_summary(
+            augmentation_summary=intent_summary, metric=metric, output_file=output_file,
+        )
+
+
+def _get_intents_with_performance_changes(
+    classification_report: Dict[Text, Dict[Text, Any]],
+    intent_report: Dict[Text, float],
```

Comment:
Please rename these arguments to reflect the difference in them - with and without augmentation